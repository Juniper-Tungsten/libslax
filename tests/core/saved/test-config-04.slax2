/* Machine Crafted with Care (tm) by slaxWriter */
version 1.1;

ns temp = "temp.example.com";

match configuration {
    <out> {
        var $name-servers = system/name-server/name;
        
        call temp:ting();
        call temp:ting($name-servers, $size = count($name-servers));
        call temp:ting($name-servers, $size = count($name-servers));
    }
}

template temp:ting ($name-servers, $size = 0) {
    <output> "template called with size " _ $size;
}
