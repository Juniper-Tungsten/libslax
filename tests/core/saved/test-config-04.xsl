<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:temp="temp.example.com" version="1.0">
  <xsl:template match="configuration">
    <out>
      <xsl:variable name="name-servers" select="system/name-server/name"/>
      <xsl:call-template name="temp:ting"/>
      <xsl:call-template name="temp:ting">
        <xsl:with-param name="name-servers" select="$name-servers"/>
        <xsl:with-param name="size" select="count($name-servers)"/>
      </xsl:call-template>
      <xsl:call-template name="temp:ting">
        <xsl:with-param name="name-servers" select="$name-servers"/>
        <xsl:with-param name="size" select="count($name-servers)"/>
      </xsl:call-template>
    </out>
  </xsl:template>
  <xsl:template name="temp:ting">
    <xsl:param name="name-servers"/>
    <xsl:param name="size" select="0"/>
    <output>
      <xsl:value-of select="concat(&quot;template called with size &quot;, $size)"/>
    </output>
  </xsl:template>
</xsl:stylesheet>
