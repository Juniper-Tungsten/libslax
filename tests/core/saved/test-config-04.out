<?xml version="1.0"?>
<out xmlns:temp="temp.example.com">
  <output>template called with size 0</output>
  <output>template called with size 1</output>
  <output>template called with size 1</output>
</out>
